cpp/src/io/parquet/compact_protocol_writer.hpp

```diff
@@ -50,6 +50,8 @@ class CompactProtocolWriter {
   size_t write(const KeyValue&);
   size_t write(const ColumnChunk&);
   size_t write(const ColumnChunkMetaData&);
+  size_t write(const PageLocation&);
+  size_t write(const OffsetIndex&);
```

Comment:
No write support for ColumnIndex?

Reply:
ColumnIndex is encoded on the device, so I didn't add it to the writer

Reply:
oh just like page headers